lib/nats.js

```diff
@@ -1184,17 +1219,37 @@ Client.prototype.request = function(subject, opt_msg, opt_options, callback) {
   if (typeof opt_msg === 'function') {
     callback = opt_msg;
     opt_msg = EMPTY;
-    opt_options = null;
+    opt_options = {};
   }
   if (typeof opt_options === 'function') {
     callback = opt_options;
-    opt_options = null;
+    opt_options = {};
+  }
+  if (!opt_options.max && opt_options.max !== false) {
+    opt_options.max = 1;
   }
   var inbox = createInbox();
+  
+  // To keep old behaviour, remove after major version bump
+  function wrapCallback (error, msg, reply) {
```

Comment:
Great that this handles the internal issue of calling the argument in a backward compatible way. But now when I go look in my IDE, I will see something that will confuse me. Because the first expected argument on the callback is an error. Yet by automagic it does something that doesn't match the signature of the api.